microsoft/react-native-test-app
# feat: add support for TurboModule

### Description

Adds support for TurboModule. I'm still seeing issues with Gradle not building things in the correct order. Until a new version of Gradle is released, we will have to point people at [Android NDK: ERROR:/~/android/src/main/jni/Android.mk:fb: LOCAL_SRC_FILES points to a missing file](https://github.com/microsoft/react-native-test-app/wiki/Troubleshooting#android-ndk-errorandroidsrcmainjniandroidmkfb-local_src_files-points-to-a-missing-file).

Resolves #836.

### Blocked by

- ~~Autolinking support on Android: https://github.com/reactwg/react-native-new-architecture/discussions/2#discussioncomment-2437670~~ Implemented in https://github.com/react-native-community/cli/pull/1603

### Platforms affected


### Test plan

- Use `react-native` 0.69: `yarn set-react-version 0.69`
- Use `@react-native-community/cli` 9.0:
   ```diff
   diff --git a/package.json b/package.json
   index c30ed18..792e470 100644
   --- a/package.json
   +++ b/package.json
   @@ -138,6 +138,9 @@
      "resolutions": {
        "@commitlint/is-ignored/semver": "^7.3.5",
        "@microsoft/eslint-plugin-sdl/eslint-plugin-react": "^7.26.0",
   +    "@react-native-community/cli": "^9.0.0-0",
   +    "@react-native-community/cli-platform-android": "^9.0.0-0",
   +    "@react-native-community/cli-platform-ios": "^9.0.0-0",
        "@react-native-community/cli-tools/shell-quote": "^1.7.3",
        "core-js-compat/semver": "^7.3.5",
        "npm/chalk": "^4.1.0"
   ```
- Re-run `yarn` after making the changes above
- Android:
  - Enable TurboModule:
    ```diff
    diff --git a/example/android/gradle.properties b/example/android/gradle.properties
    index 3cd810b..b19e049 100644
    --- a/example/android/gradle.properties
    +++ b/example/android/gradle.properties
    @@ -36,7 +36,7 @@ android.enableJetifier=true

     # Enable new architecture, i.e. Fabric + TurboModule - implies USE_FABRIC=1.
     # Note that this is incompatible with web debugging.
    -#newArchEnabled=true
    +newArchEnabled=true

     # Uncomment the line below if building react-native from source
     #ANDROID_NDK_VERSION=21.4.7075529
    ```
  - Build: `yarn android`
- iOS:
  - Install Pods: `RCT_NEW_ARCH_ENABLED=1 pod install`
  - Build: `yarn ios`
